### ISSUE TITLE
2D joints show

### ISSUE THREAD

**Issue by yyvhang** [IDEA-Research/OSX#23]
Hi, sorry to bother you, but when I run the demo.py and get the output['smplx_joint_proj'], I show the joints with the function vis_keypoints() in vis.py, but I got the following result:
<img width="537" alt="image" src="https://user-images.githubusercontent.com/94104185/233543583-9be52a56-b87b-4981-9913-9c0f36ca5d81.png">
can you tell me what's wrong whit this Situation？

**Comment by szm0613** [comment]
I also have this problem...

**Comment by yyvhang** [comment]
> I also have this problem...

This doesn't seem to have an impact on the prediction of the final mesh. However, when fitting the SMPL-X for the images that only have 2D points, like the coco-whole body, this may impact the weakly-supervised 2D loss, during your training (Fig.4), is there any problem caused by such a situation?

**Comment by linjing7** [comment]
Hi, `output['smplx_joint_proj']` is in the `output_hm_shape` plane, so you need to rescale it before showing it on the image.

**Comment by yyvhang** [comment]
> Hi, `output['smplx_joint_proj']` is in the `output_hm_shape` plane, so you need to rescale it before showing it on the image.

Many thanks for your reply, I check the following code in OSX.py.

> x = (joint_cam[:, :, 0] + cam_trans[:, None, 0]) / (joint_cam[:, :, 2] + cam_trans[:, None, 2] + 1e-4) * cfg.focal[0] + cfg.princpt[0]
y = (joint_cam[:, :, 1] + cam_trans[:, None, 1]) / (joint_cam[:, :, 2] + cam_trans[:, None, 2] + 1e-4) * cfg.focal[1] + cfg.princpt[1]
x = x / cfg.input_body_shape[1] * cfg.output_hm_shape[2]
y = y / cfg.input_body_shape[0] * cfg.output_hm_shape[1]

And I use the following code to get the 2D points

> joints = out['smplx_joint_proj'].detach().cpu().numpy()[0]
joints[:,0] = (joints[:,0] * cfg.input_body_shape[1] / cfg.output_hm_shape[2])
joints[:,1] = (joints[:,1] * cfg.input_body_shape[0] / cfg.output_hm_shape[1])

this is the result:
<img width="610" alt="image" src="https://user-images.githubusercontent.com/94104185/233757040-d964514f-edb5-42e4-b7b9-4e681116fec2.png">
It seems that these points still need to be translated. May I ask which variable is used for this translation? I tried "cam_trans", but it doesn't seem to be the correct one.

**Comment by szm0613** [comment]
hello, I try to change code like this and maybe get the correct 2D joint show

`joint_proj = out['smplx_joint_proj'].detach().cpu().numpy()[0]

joint_proj[:, 0] = joint_proj[:, 0] / cfg.output_hm_shape[2] * cfg.input_img_shape[1]
joint_proj[:, 1] = joint_proj[:, 1] / cfg.output_hm_shape[1] * cfg.input_img_shape[0]
joint_proj = np.concatenate((joint_proj, np.ones_like(joint_proj[:, :1])), 1)
joint_proj = np.dot(bb2img_trans, joint_proj.transpose(1, 0)).transpose(1, 0)`

**Comment by yyvhang** [comment]
> hello, I try to change code like this and maybe get the correct 2D joint show
> 
> `joint_proj = out['smplx_joint_proj'].detach().cpu().numpy()[0]
> 
> joint_proj[:, 0] = joint_proj[:, 0] / cfg.output_hm_shape[2] * cfg.input_img_shape[1] joint_proj[:, 1] = joint_proj[:, 1] / cfg.output_hm_shape[1] * cfg.input_img_shape[0] joint_proj = np.concatenate((joint_proj, np.ones_like(joint_proj[:, :1])), 1) joint_proj = np.dot(bb2img_trans, joint_proj.transpose(1, 0)).transpose(1, 0)`

I have tried this code, it works, just say thanks to you.
